File: ExpenseTracker.java
import java.util.ArrayList;
import java.util.Scanner;

class Expense {
    String category;
    double amount;

    Expense(String category, double amount) {
        this.category = category;
        this.amount = amount;
    }
}

public class ExpenseTracker {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        ArrayList<Expense> expenses = new ArrayList<>();

        int choice;

        do {
            System.out.println("\n===== EXPENSE TRACKER =====");
            System.out.println("1. Add Expense");
            System.out.println("2. View Expenses");
            System.out.println("3. Show Total");
            System.out.println("4. Exit");
            System.out.print("Enter Choice: ");

            choice = sc.nextInt();
            sc.nextLine();

            switch (choice) {

                case 1:
                    System.out.print("Enter Category: ");
                    String category = sc.nextLine();

                    System.out.print("Enter Amount: ");
                    double amount = sc.nextDouble();

                    if (amount > 0) {
                        expenses.add(new Expense(category, amount));
                        System.out.println("Expense Added Successfully!");
                    } else {
                        System.out.println("Invalid Amount!");
                    }
                    break;

                case 2:
                    if (expenses.isEmpty()) {
                        System.out.println("No Expenses Available.");
                    } else {
                        System.out.println("\n----- Your Expenses -----");

                        for (int i = 0; i < expenses.size(); i++) {
                            Expense e = expenses.get(i);

                            System.out.println(
                                (i + 1) + ". " +
                                e.category + " - ₹" +
                                e.amount
                            );
                        }
                    }
                    break;

                case 3:
                    double total = 0;

                    for (Expense e : expenses) {
                        total += e.amount;
                    }

                    System.out.println("Total Expenses: ₹" + total);
                    break;

                case 4:
                    System.out.println("Thank You for using Expense Tracker!");
                    break;

                default:
                    System.out.println("Invalid Choice!");
            }

        } while (choice != 4);

        sc.close();
    }
}

File: expense_tracker.py
expenses = []

while True:
    print("\n===== EXPENSE TRACKER =====")
    print("1. Add Expense")
    print("2. View Expenses")
    print("3. Show Total")
    print("4. Exit")

    choice = input("Enter your choice: ")

    if choice == "1":
        category = input("Enter category: ")
        amount = float(input("Enter amount: "))

        expenses.append({
            "category": category,
            "amount": amount
        })

        print("Expense added successfully!")

    elif choice == "2":
        if not expenses:
            print("No expenses available.")
        else:
            print("\n----- Expenses -----")

            for i, expense in enumerate(expenses, start=1):
                print(
                    f"{i}. {expense['category']} - ₹{expense['amount']:.2f}"
                )

    elif choice == "3":
        total = sum(expense["amount"] for expense in expenses)

        print(f"\nTotal Expense: ₹{total:.2f}")

    elif choice == "4":
        print("Thank you!")
        break

    else:
        print("Invalid choice!")